meilisearch-release-changelog/v1.2.0.md

```diff
@@ -0,0 +1,203 @@
+```
+
+And since many of you had really large filters to send that would definitely not fit in an URL, we also crafted a new route that you can access with the HTTP `POST` verb with the same parameter. It's located under `/indexes/:uid/documents/fetch` and can be accessed with the `limit`, `offset`, `fields`, and `filter` parameters.
+
+Here is an example of usage:
+
+```bash
+curl -X POST http://localhost:7700/indexes/dogs/documents/fetch \
+  -H 'Content-Type: application/json' \
+  --data-binary '{ "limit": 1, "filter": "doggo = bernese" }'
+```
+
+As for the previous point, the fields you're filtering on must be set as [filterable](https://www.meilisearch.com/docs/reference/api/settings#update-filterable-attributes).
+
+Done by @dureuill and @irevoire in #3570.
+
+## New filter operators: `IS EMPTY` and `IS NULL`
+
+During the search, you can use [filtering](https://www.meilisearch.com/docs/reference/api/search#filter) and retrieve documents containing `null` or empty fields.
+
+For example with the following set of documents:
+
+```json
+[{
+    "id": 0,
+    "color": []
+},
+{
+    "id": 1,
+    "color": null
+},
+{
+    "id": 2,
+    "color": ""
+},
+{
+    "id": 3,
+    "color": {}
+},
+{
+    "id": 4
+}]
+```
+
+The filter `color IS EMPTY` selects the document ids `0`, `2` and `3` while the filter `color IS NOT EMPTY` or `NOT color IS EMPTY` selects the document ids `1` and `4`.
+
+The filter `color IS NULL` selects the document ids `1` while the filter `color IS NOT NULL` or `NOT color IS NULL` selects the document ids `0`, `2`, `3` and `4`.
+
+Done by @Kerollmops in #3571.
+
+## Improve relevancy and search performance
+
+v1.2 brings a massive refactor of the search engine that brings perforance and relevancy improvements. Also, one of the major benefits of this refactoring is the ability to explore and integrate new features related to the search in the future releases. Stay tuned!
+
+Most of the points linked below can be really technical. If you have any questions or unexpected behaviors on your side, feel free to [reach us](https://github.com/meilisearch/meilisearch/discussions).
+
+Done by @loiclec, @dureuill, @ManyTheFish, @Kerollmops and @irevoire in #3542.
+
+### Performance improvements
+
+- The fastest 75 percent of queries now consistently answer below 50ms on our test setup.
+- The complexity of a search query is now limited, in order to limit CPU time and RAM consumption. In more details:
+  - The number of typo considered for a single term is now limited to 150 for 1 typo, 50 for 2 typos. Similarly, the number of considered (single word or multi word) synonyms is limited to 50, the total number of words for all synonyms of a term is limited to 100.
+  - The number of words considered in a query is now limited to 1000. The number of terms (words not inside a phrase, and individual phrases) was and is still limited to 10, so this change only impacts queries containing very long phrases.
```

Comment:
@loiclec @dureuill, what was the previous limitation?

Reply:
None

Reply:
I'm not sure to understand what it means then

Reply:
This is correct @guimachiavelli 

The meaningful difference is that very long phrases now result in the query being truncated. Even this is a detail unlikely to have much consequence in practice. I'm guessing it should maybe be documented at the same place as where we document the "10 terms" limit?